src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java

```diff
@@ -4870,6 +4870,19 @@ public void testCreateService() throws Exception {
   }
 
   @Test
+  public void testCreateServiceWithWarnings() throws Exception {
+    requireDockerApiVersionAtLeast("1.25", "swarm support");
+
+    final ServiceSpec spec = createBadServiceSpec(randomName());
+
+    final ServiceCreateResponse response = sut.createService(spec);
+    assertThat(response.id(), is(notNullValue()));
+    assertThat(response.warnings(), is(hasSize(1)));
```

Comment:
It might be nice to also have an assertion about the contents of the warning. Like, if the warning message contains the name that wasn't found, something like `assertThat(warning.get(0), contains(whatever the random name was))`. But if you don't actually know what the warning message would be, then feel free to ignore this request.